src/ImageLoaders/IHex32/IntelHex32Loader.cs

```diff
@@ -0,0 +1,240 @@
+                        if (currMemChunk == null)
+                        {
+                            currMemChunk = new MemChunk(currAddr);
+                            memChunks.Add(currAddr, currMemChunk);
+                        }
+                    }
+                }
+
+                currMemChunk.Datum.AddRange(data);
+                nextAddr = currAddr + data.Length;
+            }
+
+            public IEnumerator<MemChunk> GetEnumerator() => memChunks.Values.GetEnumerator();
+
+            IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+        }
+
+        #endregion
+
+        #region Locals
+
+        private DecompilerEventListener listener;
+
+        #endregion
+
+        #region Constructors
+
+        public IntelHex32Loader(IServiceProvider services, string filename, byte[] imgRaw)
+            : base(services, filename, imgRaw)
+        {
+        }
+
+        #endregion
+
+        #region ImageLoader interface implementation
+
+        /// <summary>
+        /// If nothing else is specified, this is the address at which the image will be loaded.
+        /// </summary>
+        /// <value>
+        /// The preferred base address.
+        /// </value>
+        public override Address PreferredBaseAddress { get; set; } = Address32.NULL;
+
+        /// <summary>
+        /// Loads the image into memory starting at the specified address.
+        /// </summary>
+        /// <param name="addrLoad">Base address of program image. IGNORED.</param>
+        /// <returns>
+        /// A <see cref="Program"/> instance.
+        /// </returns>
+        public override Program Load(Address addrLoad)
+        {
+            listener = Services.RequireService<DecompilerEventListener>();
+            MemoryChunksList memChunks = new MemoryChunksList();
+
+            using (var rdr = new IntelHex32Reader(new MemoryStream(RawImage)))
+            {
+                try
+                {
+                    for (; ; )
+                    {
+                        if (!rdr.Read(out uint address, out byte[] data))
+                            break;
+                        if (data != null)
+                        {
+                            memChunks.AddData(address, data);
+                        }
+                    }
+
+                }
+                catch (IntelHex32Exception ex)
+                {
+                    listener.Error(new NullCodeLocation(""), ex.Message);
+                }
+            }
+
+            var segs = new SegmentMap(PreferredBaseAddress);
+
+            int i = 0;
+
+            // Generate the image segments with fake names.
+            foreach (var mchk in memChunks)
+            {
+                var mem = new MemoryArea(mchk.BaseAddress, mchk.Datum.ToArray());
+                var seg = new ImageSegment($"CODE_{i++:d2}", mem, AccessMode.ReadExecute);
+                segs.AddSegment(seg);
+            }
+
+            return new Program(segs, null, null);
```

Comment:
Setting `null` architecture and platform is not a good idea. I think `Load(Address addrLoad)` should be left unimplemented and this code could be moved to `Load(Address addrLoad, IProcessorArchitecture arch, IPlatform platform)`

Reply:
Shall change. But may I ask then why this abstract method is ever proposed in ImageLoader?

Reply:
- If you use `File->Open...` menu `Reko` chooses `Image Loader` based on file extension or magic number at the start of binary. Then it call `Load(Address addrLoad)` and `Image Loader` implementation should determine arch/env based on contents of file. See `Pe Image Loader`
- If you use `File->Open As...` menu `Reko` chooses `Image Loader` based on user selection.  Then it call `Load(Address addrLoad, IProcessorArchitecture arch, IPlatform platform)`

Reply:
Clearly this needs to be documented better. I'm travelling today but hopefully I can look at this on Sunday.